app/src/main/kotlin/com/hippo/ehviewer/EhDB.kt

```diff
@@ -259,28 +260,10 @@ object EhDB {
         db.filterDao().update(filter)
     }
 
-    suspend fun exportDB(context: Context, uri: Uri) {
-        val ehExportName = "eh.export.db"
-        resource {
-            context.deleteDatabase(ehExportName)
-            roomDb<EhDatabase>(ehExportName)
-        } release {
-            it.close()
-            context.deleteDatabase(ehExportName)
-        } use { newDb ->
-            db.galleryDao().list().let { newDb.galleryDao().insertOrIgnore(it) }
-            db.progressDao().list().let { newDb.progressDao().insertOrIgnore(it) }
-            db.downloadLabelDao().list().let { newDb.downloadLabelDao().insert(it) }
-            db.downloadDirnameDao().list().let { newDb.downloadDirnameDao().insertOrIgnore(it) }
-            db.downloadsDao().list().let { newDb.downloadsDao().insert(it) }
-            db.historyDao().list().let { newDb.historyDao().insertOrIgnore(it) }
-            db.quickSearchDao().list().let { newDb.quickSearchDao().insert(it) }
-            db.localFavoritesDao().list().let { newDb.localFavoritesDao().insertOrIgnore(it) }
-            db.filterDao().list().let { newDb.filterDao().insert(it) }
-            newDb.close()
-            val dbFile = context.getDatabasePath(ehExportName)
-            dbFile.asUniFile() sendTo uri.asUniFile()
-        }
+    fun exportDB(context: Context, uri: Uri) {
+        db.query("PRAGMA wal_checkpoint(FULL)", null).use { it.moveToNext() }
+        val dbFile = context.getDatabasePath(DB_NAME)
+        dbFile.asUniFile() sendTo uri.asUniFile()
```

Comment:
Maybe we should close that db first then we can do `sendTo`
I'm afraid in-memory changes haven't been written-back to file so we will lost it.

Reply:
> In [WAL](https://www.sqlite.org/wal.html) mode when synchronous is NORMAL (1), the WAL file is synchronized before each [checkpoint](https://www.sqlite.org/wal.html#ckpt)